lib/linguist/heuristics.yml

```diff
@@ -113,6 +113,12 @@ disambiguations:
     #  : dependency
     # path/file.ext1 : some/path/../file.ext2
     pattern: '([\/\\].*:\s+.*\s\\$|: \\$|^[ %]:|^[\w\s\/\\.]+\w+\.\w+\s*:\s+[\w\s\/\\.]+\w+\.\w+)'
+- extensions: ['.dsp']
+  rules:
+  - language: Microsoft Developer Studio Project
+    pattern: '#.*Microsoft Developer Studio.*'
+  - language: Faust
+    pattern: '='
```

Comment:
This pattern is too general. Almost all languages use `=` for assignment, including `Microsoft Developer Studio Project`. Any other pattern, specific to Faust, we could use? It doesn't need to match *all* Faust files, we have other strategies to distinguish the remaining files.

Reply:
I have collected a set of regular expressions which I think relevant.
All are optional but some will match most times.

1. the main function known as `process`
2. importing statements : `import` or `library`, followed by paren and double-quote
3. some common top-level declarations

Is it good?

```
\bprocess\b
\b(library|import)\s*\(\s*"
\bdeclare\s+(name|version|author|copyright|license)\s+"
```

Reply:
The last two look good (although I don't think the `\s*` at the end of the second is necessary and the `\s+` could be `\b` for the last), but the first one is too common a keyword. Maybe we could tie that keyword to part of the construct of a function (kinda of like you did for the last two)?

Reply:
>The last two look good (although I don't think the \s* at the end of the second is necessary and the \s+ could be \b for the last)

These spacings precede a `"` character following. I intended this double-quote as part of the regular expression.

Regarding `process`, I'm able to check for the next token, which is necessarily "(" or "=".
examples are `process = _;` and `process(a,b) = a,b;`